### PR TITLE
FluxMonitor: buffer received logs

### PR DESCRIPTION
**Pull request by spooktheducks**

### CODE REVIEW

File: core/utils/utils.go [hunk 1]
```diff
@@ -520,3 +522,111 @@ func Uint256ToHex(n *big.Int) (string, error) {
...
+	}
+}
+
+func (q *BoundedQueue) Add(x interface{}) {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+	q.items = append(q.items, x)
+	if uint(len(q.items)) > q.capacity {
+		excess := uint(len(q.items)) - q.capacity
+		q.items = q.items[excess:]
```
Comment (se3000) [inline core/utils/utils.go:546]: Could this excess ever be more than 1? Do we need this if block? Maybe slightly less efficient if we trim every time, even when excess equals zero.

Comment (spooktheducks) [reply core/utils/utils.go:546]: Currently true for FM, but I want to push us to rely on more reusable, generalizable components that we can apply to other parts of the codebase.  Less testing and better reliability that way.  I've gotten a lot of mileage out of the ones we already have, most recently `BackoffSleeper`.

Re: performance, at present, we don't receive more than a few dozen logs per second, so eliminating this seems unlikely to make an impact (a comparison and two arithmetic ops).  However, with components like these, we should absolutely do regular performance profiling with `pprof` to ensure that they don't ever become bottlenecks.

File: core/utils/utils.go [hunk 2]
```diff
@@ -581,3 +582,112 @@ func (da *dependentAwaiter) AddDependents(n int) {
...
+}
+
+type BoundedPriorityQueue struct {
+	queues     map[uint]*BoundedQueue
+	priorities []uint
+	capacities map[uint]uint
+	mu         *sync.RWMutex
+}
+
+func NewBoundedPriorityQueue(capacities map[uint]uint) *BoundedPriorityQueue {
```
Comment (se3000) [inline core/utils/utils.go:640]: Can we get some tests for this?

Comment (spooktheducks) [reply core/utils/utils.go:640]: Sorry, appears the test got swallowed by a bad rebase.  Was able to recover it from my undo buffer.  Pushed.

**Pull request merged**